### PR TITLE
Add e2e docker-compose for quick start

### PR DESCRIPTION
**Pull request by VineethReddy02**
## Description

Quick start docker-compose to start all the required components alongside Promscale to get the e2e experience with Promscale.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:
* The demo applications images need to be pushed to the timescale registry.
* The APM dashboards aren't working and need some engineering. 

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

File: docker-compose/promscale-demo/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    environment:
+      SPAN_STORAGE_TYPE: grpc-plugin
+    command: [
+      "--grpc-storage.server=promscale:9202",
+    ]
+    ports:
+      - "16686:16686"
+
+  grafana:
+    image: vineeth97/promscale-demo-grafana
```
[the remark this answers is not in the document]
Comment (paulfantom) [reply docker-compose/promscale-demo/docker-compose.yaml:73]: Who will manage this image in the long run?

Is the team fine with creation of tech-debt in form of such image and if so is there a plan to switch back to "official" grafana image in the long run? If there is such plan, can we have an issue referenced in the code comment?

[the remark this answers is not in the document]
Comment (VineethReddy02) [reply docker-compose/promscale-demo/docker-compose.yaml:73]: My plan is to publish my image as `timescale/promscale-demo-grafana`. As this image will contain all the required dashboards and data sources pre-configured. The same is added to the otel-demo repo [here](https://github.com/timescale/opentelemetry-demo/pull/21).

**Review by paulfantom — COMMENTED** [review]
From technical POV this lgtm. My only concern is about using a grafana image that is not an official one.

**Review by paulfantom — APPROVED** [review]
From technical POV this lgtm. My only concern is about using a grafana image that is not an official one.

**Review by antekresic — APPROVED** [review]
lgtm

File: docker-compose/promscale-demo/docker-compose.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (alejandrodnm) [reply docker-compose/promscale-demo/docker-compose.yaml:73]: This is still in the repo as `vineeth97/promscale-demo-grafana` it's not a good look specially if we want customers to run this on their machines.

**Pull request merged**